nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateCsv.java

```diff
@@ -0,0 +1,408 @@
+
+        case "strregex":
+            return new StrRegEx(argument.substring(1, argument.length() - 1));
+
+        case "unique":
+            return new Unique();
+
+        case "uniquehashcode":
+            return new UniqueHashCode();
+
+        case "strlen":
+            String[] splts = argument.split(",");
+            int[] requiredLengths = new int[splts.length];
+            for(int i = 0; i < splts.length; i++) {
+                requiredLengths[i] = Integer.parseInt(splts[i]);
+            }
+            return new Strlen(requiredLengths);
+
+        case "strminmax":
+            String[] splits = argument.split(",");
+            return new StrMinMax(Long.parseLong(splits[0]), Long.parseLong(splits[1]));
+
+        case "lminmax":
+            String[] args = argument.split(",");
+            return new LMinMax(Long.parseLong(args[0]), Long.parseLong(args[1]));
+
+        case "dminmax":
+            String[] doubles = argument.split(",");
+            return new DMinMax(Double.parseDouble(doubles[0]), Double.parseDouble(doubles[1]));
+
+        case "equals":
+            return new Equals();
+
+        case "forbidsubstr":
+            String[] forbiddenSubStrings = argument.split(",");
+            for (int i = 0; i < forbiddenSubStrings.length; i++) {
+                forbiddenSubStrings[i] = forbiddenSubStrings[i].substring(1, forbiddenSubStrings[i].length() - 1);
+            }
+            return new ForbidSubStr(forbiddenSubStrings);
+
+        case "requiresubstr":
+            String[] requiredSubStrings = argument.split(",");
+            for (int i = 0; i < requiredSubStrings.length; i++) {
+                requiredSubStrings[i] = requiredSubStrings[i].substring(1, requiredSubStrings[i].length() - 1);
+            }
+            return new RequireSubStr(requiredSubStrings);
+
+        case "strnotnullorempty":
+            return new StrNotNullOrEmpty();
+
+        case "requirehashcode":
+            String[] hashs = argument.split(",");
+            int[] hashcodes = new int[hashs.length];
+            for(int i = 0; i < hashs.length; i++) {
+                hashcodes[i] = Integer.parseInt(hashs[i]);
+            }
+            return new RequireHashCode(hashcodes);
+
+        case "null":
+            return null;
+
+        default:
+            throw new IllegalArgumentException(method + " is not an allowed method to define a Cell Processor");
+        }
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+        final List<FlowFile> flowFiles = session.get(50);
+        if (flowFiles.isEmpty()) {
+            return;
+        }
+
+        final CsvPreference csvPref = this.preference.get();
+        final boolean header = context.getProperty(HEADER).asBoolean();
+        final ComponentLog logger = getLogger();
+        final CellProcessor[] cellProcs = this.processors.get().toArray(new CellProcessor[this.processors.get().size()]);
```

Comment:
Can this be done in the onScheduled? I feel there is decent amount of allocating and processing that could be re-used.